### DISCUSSION TITLE
How to get CSI from ray tracing?

### DISCUSSION THREAD

**Discussion by remydemy** [NVlabs/sionna#135] [Q&A] [answered]
Hello,

I've modelled a scene with one transmitter and one receiver, computed their paths and calculated the ofdm channel with `cir_to_ofdm_channel()`.

My goal now is to get some information at the transmitter and receiver so that I can calculate a symmetric key at both ends. It doesn't really matter if I get the info from CSI or some pilot signal or something else. Do I need to send something between them before I can access such data? If so how can I do this? Or am I on a completely wrong path?

What I also don't understand is that is the h that I calculate with `cir_to_ofdm_channel()` different for the receiver and the transmitter, because they are at different positions in the scene? If so how can I access these different values, because I'm only using this method once (with the a, tau that I got from `Paths2CIR`).

Thanks for any help!

Best regards,
Emily

**Comment by jhoydis** [comment, marked as answer]
Hi Emily,

When you compute the CIR between two radio devices via ray tracing, it should not matter which device is the transmitter and which is the receiver. The CIR should be the same. It is therefore not necessary to compute it twice.

In practice, none of the radio devices has access to the CIR and will only get an estimate of it, based on pilots sent by the other radio device. To simplify this approach, you could compute the channel frequency response h once using ``cir_to_ofdm_channel()`` and then create two noisy versions of it by adding two different random noise vectors to h.

Hope that helps.

**Reply by remydemy** [reply]
Thanks for your fast response!
I assume adding `AWGN()` to simulate noise would be sufficient for this purpose?

**Reply by jhoydis** [reply]
Yes, you can use this layer, or alternatively use the convenience function [``sionna.utils.complex_normal()``](https://nvlabs.github.io/sionna/api/utils.html#sionna.utils.complex_normal). The variance parameter will determine the quality of your channel estimates. I believe that you should probably set the `normalise=True` flag in the ``cir_to_ofdm_channel()`` function such that the channel frequency response has unit power on average.

**Reply by remydemy** [reply]
Ok great. I actually already set `normalise=True`.

You were really helpful!

Kind regards,
Emily

**Reply by fighting-xq** [reply]
How to using AWGN() in sionna RT?
